.github/workflows/pr-create-release-build.yml

```diff
@@ -0,0 +1,199 @@
+        run: |
+         import sys
+         import subprocess
+         tag_name = "${{ github.event.inputs.tag_name }}"
+         command = ['git', 'tag', '-l', tag_name]
+         output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+         if output.decode() != "":
+           print(f"Error: Tag '{tag_name}' already exists.", file=sys.stderr)
+           sys.exit(1)
+         else:
+           print(f"Tag '{tag_name}' does not exists.")
+        
+        shell: python
+        continue-on-error: false
+
+  create-pr:
+    runs-on: ubuntu-latest
+    needs: check-prev-tag
+    env:
+      GITHUB_BRANCH: ${{ github.ref }}
+    outputs:
+      pr_number: ${{ steps.create-pull-request.outputs.pr_number }} 
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+      - name: Create and checkout new branch
+        id: create_branch
+        run: |
+          BRANCH_NAME="update-param-env-${{ github.event.inputs.tag_name }}"
+          echo "BRANCH_NAME=${BRANCH_NAME}" >> $GITHUB_ENV
+          git checkout -b $BRANCH_NAME
+      - name: Update params.env with new release version
+        run: |     
+          sed -i 's|:v[0-9.]*\b|:${{ github.event.inputs.tag_name }}|gm' config/overlays/odh/params.env
+      - name: Commit changes
+        run: |
+          git add config/overlays/odh/params.env
+          git commit -m "Update image refs for odh release"
+          git push origin $BRANCH_NAME
+      - name: Create Pull Request
+        id: create-pull-request
+        run: | 
+          PR_URL=$(gh pr create -B ${{ github.ref }} -H ${{ env.BRANCH_NAME }} --title '[ODH Release] Update images for ${{ github.event.inputs.tag_name }}' --body 'Update images for ${{ github.event.inputs.tag_name }}')
+          echo "PR_URL=${PR_URL}" >> $GITHUB_ENV
+          pr_number=$(echo "$PR_URL" | grep -o -E '[0-9]+$')
+          echo "pr_number=${pr_number}" >> $GITHUB_OUTPUT
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+  wait-checks:
+    runs-on: ubuntu-latest
+    needs: [ check-prev-tag,create-pr ] 
+    steps:
+      - name: Watching PR if Checks finished without errors
+        id: wait-checks
+        run:
+          gh pr checks  ${{ needs.create-pr.outputs.pr_number }}  --watch --fail-fast 
+        env:     
+         GH_TOKEN: ${{ github.token }}
+ 
+  rename-image:
+    needs: [ check-prev-tag,create-pr,wait-checks]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to Quay.io
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.Q_PASSWORD }}

```

Comment:
Should it be QUAY_PASSWORD?

Reply:
Yeah i have changed it even i am not sure of the variable name as i have no access to review the settings in the repo so i have reffered this file[LINK]( https://github.com/opendatahub-io/kserve/blob/master/.github/workflows/kserve-controller-docker-publish.yml)

Reply:
In the configs, I see it is QUAY_USER and QUAY_PASSWORD.
So, you may want to change it to QUAY_USER, rather than QUAY_USERNAME.

I do not have access to the values, so I'm not sure if it can be renamed... Although, I should have the credentials stored somewhere... So, we may merge as it is and I can later setup the new variable name.